lib/json_api_resource/queryable.rb

```diff
@@ -35,6 +37,19 @@ def where(opts = {})
         pretty_error e
       end
 
+      def all_without_pagination(opts = {})
+        page_total = 1
+        current_page = 1
+        all_results = []
+        until (current_page > page_total) || (current_page > MAX_PAGES_FOR_ALL)
+          page_of_results = where({:page => current_page}.merge(opts))
+          all_results << page_of_results
+          page_total = page_of_results.meta[:total_pages]
+          current_page = current_page + 1
+        end
+        all_results.flatten.compact
```

Comment:
what if we wrap this in JsonApiClient::ResultSet so the output is the same across all query methods?